### PR TITLE
Introduce debian folder

### PR DESCRIPTION
**Pull request by mhs4670go**
This commit introduces debian folder for generating debian package.

Related: #6717 
ONE-DCO-1.0-Signed-off-by: seongwoo <mhs4670go@naver.com>

### CODE REVIEW

File: debian/copyright [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Files: *
+License: Proprietary
+Copyright (c) <2018> <Samsung Electronics Co.,Ltd.>
```
Comment (seanshpark) [inline debian/copyright:3]: 2018 is tart of _nnfw_ ?

File: debian/control [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+Package: one-compiler
+Architecture: any
+Multi-Arch: foreign
+Depends: ${misc:Depends}, ${shlibs:Depends}
+Description: on-device neural engine compiler
+
+Package: one-compiler-dev
+Architecture: any
+Multi-Arch: same
+Depends: ${shlibs:Depends}, ${misc:Depends}
```
Comment (seanshpark) [inline debian/control:18]: Can `one-compiler-dev` work without `one-compiler` ? If not, we need to add depend on `one-compiler`

File: debian/postinst [hunk 3]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+
+# https://www.debian.org/doc/debian-policy/ch-maintainerscripts.html
+# Boradly speaking, the `postinst` is called after a package is unpacked.
+
+set -e
+
+/opt/one_compiler/bin/one-prepare-venv
```
Comment (seanshpark) [inline debian/postinst:8]: If user can change installation path to somewhere else, is `/opt/one_compiler/bin/one-prepare-venv` available?

File: debian/copyright (hunk 1 above)
Comment (mhs4670go) [reply debian/copyright:3]: @seanshpark Right.

File: debian/control (hunk 2 above)
Comment (mhs4670go) [reply debian/control:18]: @seanshpark I think `one-compiler-dev` should work without `one-compiler`. I'll add `lib` to its install list.

File: debian/postinst (hunk 3 above)
Comment (mhs4670go) [reply debian/postinst:8]: @seanshpark Umm I thought user can't change the installation path. Is it possible?

File: debian/control (hunk 2 above)
Comment (seanshpark) [reply debian/control:18]: > I'll add lib to its install list.

Does this mean, both `one-compiler` and `one-compiler-dev` will install `lib` ?

File: debian/postinst (hunk 3 above)
Comment (seanshpark) [reply debian/postinst:8]: > Is it possible?

I don't know either...
But with some searching, it seems not possible :)

File: debian/control (hunk 2 above)
Comment (seanshpark) [reply debian/control:18]: > Yes. That's what I meant to.

I think this may make conflict on `lib`.

Comment (mhs4670go) [reply debian/control:18]: > I think `one-compiler-dev` should have a dependency on `one-compiler`.

Is it for preventing `one-compiler-dev` from having duplicate `lib` list?

**Comment by mhs4670go** [comment]
[one-compiler.tar.gz](https://github.com/Samsung/ONE/files/6449902/one-compiler.tar.gz)
This is the package from this PR.

File: debian/control [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+Maintainer: Neural Network Acceleration Solution Developers <nnfw@samsung.com>
+Build-Depends: cmake, debhelper (>=9), dh-python, python3-all
+Standards-Version: 3.9.8
+Homepage: https://github.com/Samsung/ONE
+
+Package: one-compiler
+Architecture: amd64
+Multi-Arch: foreign
+Depends: ${misc:Depends}, ${shlibs:Depends}
+Description: on-device neural engine compiler
```
Comment (seanshpark) [inline debian/control:13]: `On-device Neural Engine compiler` ?

Comment (seanshpark) [reply debian/control:13]: maybe `Description: ... compiler package` ? as below two ends with `package`

**Review by seanshpark — APPROVED** [review]
LGTM thank you!

**Review by chunseoklee — APPROVED** [review]

File: debian/one-compiler-dev.install [hunk 5]
```diff
@@ -0,0 +1,5 @@
+# {FILES_TO_INSTALL} {DEST_DIR}
+# bin
+usr/bin/circledump opt/one_compiler/bin/
```
[the remark this answers is not in the document]
Comment (seanshpark) [reply debian/one-compiler-dev.install:3]: `/usr/local/share/one` :)

[the remark this answers is not in the document]
Comment (jyoungyun) [reply debian/one-compiler-dev.install:3]: My mistake, Thank you :)

**Pull request merged**